dynamic_programming/box_stacking.cpp

```diff
@@ -0,0 +1,95 @@
+/* Dynamic Programming implementation of Box Stacking problem */
```

Comment:
Please place a longer overview that conforms to the [Doxygen standards](https://www.doxygen.nl/manual/docblocks.html) here, check the example in the [contribution guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#typical-structure-of-a-program)